i-VRESSE/workflow-builder
# Make node schema maxItemsFrom and molecule aware

Refs #59

To test on small catalog
1. Start app
2. select the `molaware` catalog
3. fill the molecules array in the global parameters with some PDB files, downloaded from https://www.rcsb.org
4. Add a node and open its form
5. The properties should have maxItems equal to the number of molecule files that where uploaded.
6. The node 3 and 4 use the PDB files to limit the allowed chains and residues in their props. A PDB file with just chain 'A' should only allow 'A' to be selected in the chain property. A PDB files with residue number 1..100 should only allow 1..100 to be selected in the residue property.

To test on haddock catalog
1. Start app
2. select the `haddock3expert` catalog
3. fill the molecules array in the global parameters with some PDB files, downloaded from https://www.rcsb.org
4. Add `rigidbody` node, edit, expand molecule
   - mol_fix_origin has max items of number of molecules
5. Add `flexref` node, edit, expand flexibility
   - seg has same number of items as molecules
   - seg start/end have residues

To test validation of upload
1. Start app
2. select the `haddock3expert` catalog
3. Upload [workflow (18).zip](https://github.com/i-VRESSE/workflow-builder/files/8553737/workflow.18.zip) which is valid
4. Upload [workflow-bad-resnum.zip](https://github.com/i-VRESSE/workflow-builder/files/8554746/workflow-bad-resnum.zip) which has bad residue number
5. For validation errors see DevTools console
